### ISSUE TITLE
sticky header offsetY not working 

### ISSUE THREAD

**Issue by kirancheema** [wenzhixin/bootstrap-table#4755]
Hello Thanks for this project - First time using it today and its really useful! 

I have been trying the sticky Header extention and it seems to be working - Unless you try and add the stickyHeaderOffsetY 

using Local host - Uniserver - apache / php and Google chrome 

Downloaded today build to 1.15.5  

Using config   
`$table.bootstrapTable({
      stickyHeader: true,
      stickyHeaderOffsetY: 0 ,
      stickyHeaderOffsetLeft: '3em',
      stickyHeaderOffsetRight: '3em',
    });`

Outputs 
  Table looks good 
![image](https://user-images.githubusercontent.com/9959732/71087042-243b5a00-2193-11ea-85bf-0839d22a5e3a.png)

On scroll - Sticky seems fine .. needs offset 

![image](https://user-images.githubusercontent.com/9959732/71087129-5056db00-2193-11ea-93b3-a54af7a69e21.png)

checking on inspect : top:0px inline style
![image](https://user-images.githubusercontent.com/9959732/71087498-39fd4f00-2194-11ea-8bf6-1bd7f265fcd2.png)
 

Update Config : 
`  $table.bootstrapTable({
      stickyHeader: true,
      stickyHeaderOffsetY: 60 ,
      stickyHeaderOffsetLeft: '3em',
      stickyHeaderOffsetRight: '3em',
    });`

Header is no longer Sticky : 

![image](https://user-images.githubusercontent.com/9959732/71087298-be9b9d80-2193-11ea-873d-2b7af05f81cf.png)

On inspect top inline style is no longer applied  

![image](https://user-images.githubusercontent.com/9959732/71087411-002c4880-2194-11ea-89d3-69458f86778b.png)

Unfortunately there is no on-line example at the moment, Just trying on local 

I have a quick fix by adding extra CSS
 
.sticky-header-container.fix-sticky.fixed-table-container {top: 60px;}

**Comment by wenzhixin** [comment]
Please provide an [Online Example](https://live.bootstrap-table.com) to show your problem, thanks!

**Comment by wenzhixin** [comment]
Any update?

**Comment by dreamerworks** [comment]
Got the same problem.
I think it is because of line 60:
`const end = this.$stickyEnd.offset().top - this.options.stickyHeaderOffsetY - this.$header.height()`
If you set it to "60px" then the result will be NaN. 
If you set it to "60" the result is well calculated but then, on line 76:
`this.$stickyContainer.css('top', `${this.options.stickyHeaderOffsetY}`)`
it doesn't work because it does not have a unit of measure (px,em,...).

**Comment by Vity01** [comment]
I can confirm @dreamerworks findings. I found same solution for that:
```
this.$stickyContainer.css('top', (this.options.stickyHeaderOffsetY) ? this.options.stickyHeaderOffsetY + "px" : '');
this.$stickyContainer.css('left', (this.options.stickyHeaderOffsetLeft) ? this.options.stickyHeaderOffsetLeft + 'px' : '');
this.$stickyContainer.css('right', (this.options.stickyHeaderOffsetRight)? this.options.stickyHeaderOffsetRight + 'px': ''); // create scrollable container for header
```

**Comment by UtechtDustin** [comment]
@dreamerworks 
@Vity01 
Could one of you provide us an example of the bug, so we can test/fix it ?

**Comment by dreamerworks** [comment]
https://live.bootstrap-table.com/code/dreamerworks/1699
can you see this?

**Comment by UtechtDustin** [comment]
@dreamerworks thank you :-)

**Comment by UtechtDustin** [comment]
@djhvscf @wenzhixin 
I dont get it, the solution of @dreamerworks almost works.
The header has the right position (y-offset), but the header will not be hidden if you scroll up.
The problem is the `sticky_anchor_begin` div, the y position of that div is much higher then the table.
![Bildschirmfoto von 2020-02-07 18-23-57](https://user-images.githubusercontent.com/13292481/74051106-a730a200-49d7-11ea-90e5-0d08ab43e15f.png)

**Comment by dreamerworks** [comment]
Attempting to revive this issue, one solution to the "px,%, em" problem would be to add another property like stickyHeaderOffsetYUnit and use it like:
`this.$stickyContainer.css('top', (this.options.stickyHeaderOffsetY) ? this.options.stickyHeaderOffsetY + this.options.stickyHeaderOffsetYUnit : '');
`
No solution for the issue @UtechtDustin described.

**Comment by Bensemer** [comment]
@dreamerworks The .offset() and the .height() function returning px, so in my opinion stickyHeaderOffsetY should always be px or you have to calc how much px the % or em is you need for the const end.

**Comment by Marcusyan** [comment]
this.$stickyContainer.css('top', "".concat(this.options.stickyHeaderOffsetY));
===>
this.$stickyContainer.css('top', "".concat(this.options.stickyHeaderOffsetY)+'px');

**Comment by syedsanahassan** [comment]
Check the comment above from @Marcusyan that shows the correct solution 

> this.$stickyContainer.css('top', "".concat(this.options.stickyHeaderOffsetY));
> ===>
> this.$stickyContainer.css('top', "".concat(this.options.stickyHeaderOffsetY)+'px');

**Comment by dreamerworks** [comment]
I would agree with the point of @Bensemer about the .offset() and the .height() function returning px. So, do go with this option to solve the bug. 
However.... if we want to be picky about it... on https://examples.bootstrap-table.com/#extensions/sticky-header.html#view-source you see:
` stickyHeaderOffsetLeft: '3em',
 stickyHeaderOffsetRight: '3em',`
which would lead to believe that stickyHeaderOffsetY should be able to defined in a similar fashion. 
And yes it would a lot of work to calculate what 3em means in px to apply it in the formula.

**Comment by UtechtDustin** [comment]
@wenzhixin i guess we have to revert #5590, i agree with @dreamerworks. It would be hard to calculate e.g. 3em in px.
Also we now have different behaviour on the offset options.

**Comment by wenzhixin** [comment]
@UtechtDustin fixed in #5591
